ci-operator/jobs/openshift/installer/openshift-installer-presubmits.yaml

```diff
@@ -90,6 +90,64 @@ presubmits:
           - configMap:
               name: cluster-profile-aws
     trigger: ((?m)^/test( all| e2e-aws),?(\s+|$))
+  - agent: kubernetes
+    always_run: false
+    branches:
+    - master
+    context: ci/prow/e2e-aws-smoke
```

Comment:
We don't want the installer repo running both `e2e-aws-smoke` and `e2e-aws`.  If we want to be able to toggle smoke tests on for openshift/installer and off for other repos, can we stick to a single `e2e-aws` (and associated template) and set a `SMOKE_TEST` environment variable non-empty just for openshift/installer?  Or something like that; I don't know what the idomatic Prow approach would be; @smarterclayton, @stevekuznetsov?

Reply:
@wking @smarterclayton I'm thinking we should merge as/is, with separate templates (will be same workflow we currently have, to run smoke tests you'll `/test e2e-aws-smoke`  except now instead of running a jenkins job smoke tests will now run here in prow.  We can look to consolidate/leverage a single cluser for both in a follow-up? Team is currently blocked waiting for this PR.

Reply:
> We can look to consolidate/leverage a single cluser for both in a follow-up?

That's fine with me.

Reply:
The structure of the pod for installer (which needs to link in the source code of the installer for the smoke test) is different for when e2e is testing itself vs when e2e is testing others.

The follow up PR (once smoke is deflated and working correctly) can replace the installer's e2e-aws run with e2e-aws-smoke.  However, anyone changing aws-smoke also needs to update the other aws template, because we will start adding the e2e-aws job to other repos to gate their merges (i.e. CVO won't merge changes unless they pass e2e-aws).

Reply:
> The structure of the pod for installer (which needs to link in the source code of the installer for the smoke test) is different for when e2e is testing itself vs when e2e is testing others.

We don't need additional source to [run the smoke tests][1] (although we do need the source to [*build* the smoke tests][2]).  Is there not a good way to toggle [the smoke-specific `initContainers`][3]?

[1]: https://github.com/openshift/release/commit/032401ade27d0b61441781f71a99a72443df788c#diff-3ff94bc11f976ad82683a65aee7f2495R256
[2]: https://github.com/openshift/release/commit/032401ade27d0b61441781f71a99a72443df788c#diff-cdc7308edc429663767c3fd839d38899R56
[3]: https://github.com/openshift/release/commit/032401ade27d0b61441781f71a99a72443df788c#diff-3ff94bc11f976ad82683a65aee7f2495R64

Reply:
No, and anyway the init containers would need an input.  Two templates isn't the end of the world in the short term (in the longer run these will be code in the ci-operator that can share more).